lib/jekyll/utils.rb

```diff
@@ -297,5 +287,36 @@ def merged_file_read_opts(site, opts)
       merged
     end
 
+    private
+    def merge_values(target, overwrite)
+      target.merge!(overwrite) do |_key, old_val, new_val|
+        if new_val.nil?
+          old_val
+        elsif mergable?(old_val) && mergable?(new_val)
+          deep_merge_hashes(old_val, new_val)
+        else
+          new_val
+        end
+      end
+    end
+
+    private
+    def merge_default_proc(target, overwrite)
+      if default_proc?(target) && default_proc?(overwrite) && target.default_proc.nil?
+        target.default_proc = overwrite.default_proc
+      end
+    end
+
+    private
+    def default_proc?(object)
```

Comment:
This method seems like it might be checking whether the object _is_ a default proc. Which is a bit confusing. Perhaps we could just check if these are `Hash`es?